app/web/features/profile/view/UserOverview.test.tsx

```diff
@@ -0,0 +1,119 @@
+import { render } from "@testing-library/react";
+import { HostingStatus, MeetupStatus } from "proto/api_pb";
+import wrapper from "test/hookWrapper";
+import { addDefaultUser } from "test/utils";
+
+import {
+  COMMUNITY_STANDING,
+  hostingStatusLabels,
+  meetupStatusLabels,
+  VERIFICATION_SCORE,
+} from "../constants";
+import { ProfileUserProvider } from "../hooks/useProfileUser";
+import UserOverview from "./UserOverview";
+
+describe("UserOverview", () => {
+  beforeEach(() => {
+    addDefaultUser();
+  });
+
+  describe("when user is missing", () => {
+    it("doesn't fail", () => {
+      render(<UserOverview />, { wrapper });
+    });
+  });
+
+  describe("when user is loaded and provided via context", () => {
+    it("should display the user name", () => {
+      const { getByText } = render(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability={false} />
+        </ProfileUserProvider>,
+        { wrapper }
+      );
+      expect(getByText(defaultUser.name)).toBeInTheDocument();
+    });
+
+    it("should display the user location", () => {
+      const { getByText } = render(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability={false} />
+        </ProfileUserProvider>,
+        { wrapper }
+      );
+      expect(getByText(defaultUser.city)).toBeInTheDocument();
+    });
+
+    it("should display the hosting and meeting status only when showHostAndMeetAvailability is true", () => {
+      const expectedLabelHosting =
+        hostingStatusLabels[defaultUser.hostingStatus as HostingStatus];
+      const expectedLabelMeeting =
+        meetupStatusLabels[defaultUser.meetupStatus as MeetupStatus];
+
+      const { queryByText, getByText, rerender } = render(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability={false} />
+        </ProfileUserProvider>,
+        { wrapper }
+      );
+      expect(queryByText(expectedLabelHosting)).not.toBeInTheDocument();
+      expect(queryByText(expectedLabelMeeting)).not.toBeInTheDocument();
+      rerender(
```

Comment:
I'd call `render` again in another test case rather than using `rerender` altogether, since it doesn't render with all of the right providers that you passed in from `wrapper` in the original `render`. It's also a bit of an escape hatch and generally should be avoided, since it doesn't encourage testing from how user would interact with the UI code.

Reply:
that makes sense. I'll update it. Thanks